### PR TITLE
[hlsl-out]: Document storage access generation.

### PR DESCRIPTION
**Pull request by jimblandy**

### CODE REVIEW

File: src/back/hlsl/mod.rs [hunk 1]
```diff
@@ -279,6 +279,25 @@ pub struct Writer<'a, W> {
...
     wrapped: Wrapped,
+
+    /// A simplified representation of a global variable access chain.
+    ///
+    /// See the [`storage`] module's documentation for background on
+    /// why we need this.
+    ///
+    /// Each [`SubAccess`] is a lowering of some [`Access`] or
+    /// [`AccessIndex`] expression to the level of byte strides and
+    /// offsets. See the [`SubAccess`] docs for details.
```
Comment (ErichDonGubler) [inline src/back/hlsl/mod.rs:290]: nitpick:

```suggestion
    /// offsets. See [`SubAccess`] for details.
```

File: src/back/hlsl/mod.rs [hunk 2]
```diff
@@ -279,6 +279,25 @@ pub struct Writer<'a, W> {
...
+    ///
+    /// To access a composite value, we need to generate a series of
+    /// loads or stores of its individual components. For that, we
+    /// temporarily push a new `SubAccess` for each component, perform
```
Comment (ErichDonGubler) [inline src/back/hlsl/mod.rs:294]: nitpick: Adding brackets makes things marginally nicer for readers WRT consistency in linking, and it's not laborious:
 
```suggestion
    /// temporarily push a new [`SubAccess`] for each component, perform
```

File: src/back/hlsl/storage.rs [hunk 3]
```diff
@@ -366,6 +414,17 @@ impl<W: fmt::Write> super::Writer<'_, W> {
...
     }
 
+    /// Set [`temp_access_chain`] to compute the byte offset of `cur_expr`.
+    ///
+    /// The `cur_expr` expression must be a reference to a global
+    /// variable in the [`Storage`] address space, or a chain of
+    /// [`Access`] and [`AccessIndex`] expressions referring to some
+    /// component of such a global.
+    ///
+    /// [`temp_access_chain`]: super::Writer::temp_access_chain
```
Comment (ErichDonGubler) [inline src/back/hlsl/storage.rs:424]: question: Wait, does this even work for fields? 👀 EDIT: YES IT DOES. 😀 Woot! I'd even been using this notation before, but I don't remember trying it with success before. TIL!

**Review by ErichDonGubler — APPROVED** [review]
LGTM! Had some non-blocking suggestions.

File: src/back/hlsl/mod.rs (hunk 2 above)
Comment (jimblandy) [reply src/back/hlsl/mod.rs:294]: Yeah... but it seems to me that links are slightly distracting, so I usually only make the first use of a term in each section a link.

File: src/back/hlsl/storage.rs [hunk 4]
```diff
@@ -1,7 +1,30 @@
...
+that we can't use HLSL array element indexing or struct member
+references to access their contents. Instead, we must emit `Load` and
+`Store` calls, and manually compute the byte offset of the value we
+need.
+
+To generate code for a [`Storage`] access:
+
+- Call [`Writer::fill_access_chain`] to prepare for access to a given
+  [`Storage`] value. This populates [`Writer::temp_access_chain`] with
+  the appropriate byte offset calculations.
```
[the remark this answers is not in the document]
Comment (ErichDonGubler) [reply src/back/hlsl/storage.rs:15]: In case it's not clear: this suggestion is intended as a suggestion for a separate PR. 😅

**Review by jimblandy — COMMENTED** [review]
Okay, I've made the requested changes.

**Pull request merged**